packer-plugin-sdk/multistep/commonsteps/step_http_server.go

```diff
@@ -54,14 +54,20 @@ func (s *StepHTTPServer) Run(ctx context.Context, state multistep.StateBag) mult
 		return multistep.ActionHalt
 	}
 
-	ui.Say(fmt.Sprintf("Starting HTTP server on port %d", s.l.Port))
+	httpAddr = state.Get("http_ip").(string)
+	if httpAddr != "" {
```

Comment:
This should be the opposite: 
```suggestion
	if httpAddr == "" {
```

Reply:
This was the right idea, but we need to swap that check -- we want to override if the address loaded from state is empty, not the other way around.

I verified that when I swap the boolean this works for the qemu builder, which was one of the ones I was concerned this change might break. 

Thanks for making this change!